test/integration/EndToEnd.t.sol

```diff
@@ -648,6 +644,40 @@ contract EndToEndUseCases is EndToEndFlows {
         assertEq(s.spoke.shareToken(POOL_A, SC_1).hook(), address(s.fullRestrictionsHook));
     }
 
+    /// forge-config: default.isolate = true
+    function testFullRefundSubsidizedCycle() public {
+        _setSpoke(false);
+        _createPool();
+
+        vm.startPrank(FM);
+        h.hub.notifyPool{value: GAS}(POOL_A, s.centrifugeId);
+        h.hub.notifyShareClass{value: GAS}(POOL_A, SC_1, s.centrifugeId, address(0).toBytes32());
+        h.hub.updateBalanceSheetManager{value: GAS}(s.centrifugeId, POOL_A, BSM.toBytes32(), true);
+        h.hub.updateSharePrice(POOL_A, SC_1, ZERO_PRICE);
+        h.hub.notifySharePrice{value: GAS}(POOL_A, SC_1, s.centrifugeId);
+        h.hub.setSnapshotHook(POOL_A, h.snapshotHook);
+
+        // Each message will return half of the gas wasted
+        adapterBToA.setRefundedValue(h.gasService.updateShares() / 2);
+
+        // We just subsidize for two message
+        vm.startPrank(ANY);
+        s.gateway.subsidizePool{value: h.gasService.updateShares() * 2}(POOL_A);
+
```

Comment:
I think this would make the test easier to follow:
```suggestion
        // We just subsidize for two message
        vm.startPrank(ANY);
        s.gateway.subsidizePool{value: h.gasService.updateShares() * 2}(POOL_A);
        assertEq(address(s.balanceSheet.escrow(POOL_A)).balance, 0);
```

Potentially also asserting the subsidized funds in the gateway. So you can see the amount of subsidized funds available going down, and the amount in the escrow going up.

Similar to test style like this: https://github.com/sparkdotfi/spark-alm-controller/blob/dev/test/mainnet-fork/Centrifuge.t.sol#L221